sys/kern/src/syscalls.rs

```diff
@@ -59,7 +59,7 @@ use crate::umem::{safe_copy, USlice};
 /// the way in and restored on the way out, (2) call this from the syscall
 /// interrupt handler, only, and (3) not call it reentrantly.
 #[no_mangle]
-pub unsafe extern "C" fn syscall_entry(nr: u32, task: *mut Task) {
+pub unsafe extern "C" fn syscall_entry(nr: u32, task: *const Task) {
```

Comment:
dumb question on my end probably: does const in rust apply to chasing pointers of other fields in the object? C/C++ does not apply that all the way through the addressability of the object.

I was asking because of potentially needing to update a task's state. However, we just take the mutable container of tasks itself and pass it along.

Reply:
Yeah, this path doesn't really pass the task pointer forward. It uses it only to calculate the index of the task in the task table and passes that to the next function.

To answer your original question, I don't think rust guarantees const on chasing pointers. In fact, my limited understanding is that this const is just for raw pointers and guarantees mutation not being allowed directly to the memory pointed by the pointer.